Snowflake.Data/Core/Session/SFSession.cs

```diff
@@ -50,7 +51,9 @@ public class SFSession
 
         internal string serverVersion;
 
-        internal TimeSpan connectionTimeout;
+        private readonly ConnectionPoolConfig _poolConfig;

```

Comment:
should we reuse SessionPool's config rather than create here a clone of it?

Reply:
The only thing which is used from the poolConfig is connectionTimeout.
We don't have a setter to change the ConnectionTimeout, we have only a setter to change ExpirationTimeout.
So cloning the configuration objects doesn't introduce any difference in behaviour, it is simpler and there are no concurrency concerns of sharing the same object in multiple places.